### PR TITLE
Added Inversify

### PR DESCRIPTION
**Pull request by josipslavic**

### CODE REVIEW

File: src/controllers/comment.controller.ts [hunk 1]
```diff
@@ -1,17 +1,20 @@
...
 import { isValidMongoId } from '../utils/isValidMongoId'
 
+@injectable()
 export class CommentController {
   constructor(
-    private commentService: CommentService,
-    private newsService: NewsService
+    @inject(TYPES.CommentService) private commentService: CommentService,
+    @inject(TYPES.NewsService) private newsService: NewsService
   ) {}
```
Comment (anteljubic) [inline src/controllers/comment.controller.ts]: You are missing binding commentService and newsService to `this` on all controllers

File: src/config/inversify.ts [hunk 2]
```diff
@@ -0,0 +1,32 @@
...
+
+container.bind<Model<INews>>(TYPES.NewsModel).toConstantValue(NewsModel)
+container.bind<NewsService>(TYPES.NewsService).to(NewsService)
+container.bind<NewsController>(TYPES.NewsController).to(NewsController)
+
+container
+  .bind<Model<IComment>>(TYPES.CommentModel)
+  .toConstantValue(CommentModel)
+container.bind<CommentService>(TYPES.CommentService).to(CommentService)
+container.bind<CommentController>(TYPES.CommentController).to(CommentController)
```
Comment (anteljubic) [inline src/config/inversify.ts:35]: Usually we group controllers, models, services etc. together, so you can regroup those bindings

File: src/types/types.ts [hunk 3]
```diff
@@ -0,0 +1,13 @@
...
+  UserController: Symbol.for('UserController'),
+
+  NewsModel: Symbol.for('NewsModel'),
+  NewsService: Symbol.for('NewsService'),
+  NewsController: Symbol.for('NewsController'),
+
+  CommentModel: Symbol.for('CommentModel'),
+  CommentService: Symbol.for('CommentService'),
+  CommentController: Symbol.for('CommentController'),
+}
```
Comment (anteljubic) [inline src/types/types.ts:17]: You can group types here as well by model, service, controller etc.

**Review by anteljubic — CHANGES_REQUESTED** [review]
Nice progress, but you can do more, you can change routes as well to be injectable

**Review by anteljubic — APPROVED** [review]

**Review by JCukovic — APPROVED** [review]

**Pull request merged**